File: src/main/java/io/awilson/timein/services/InstructorService.java
package io.awilson.timein.services;

import io.awilson.timein.domain.Instructor;

public interface InstructorService {
    Iterable<Instructor> listAllInstructors();

    Instructor getInstructorById(Integer id);

    Instructor saveInstructor(Instructor instructor);

    void deleteInstructor(Integer id);
}

File: src/main/java/io/awilson/timein/services/InstructorService.kt
package io.awilson.timein.services

import io.awilson.timein.domain.Instructor

interface InstructorService {
    fun listAllInstructors(): Iterable<Instructor>

    fun getInstructorById(id: Int): Instructor?

    fun saveInstructor(instructor: Instructor): Instructor

    fun deleteInstructor(id: Int)
}
